week-3/stretch/rotate-char.js

```diff
@@ -6,6 +6,33 @@
 // This function is commonly used for weak text encryption and decryption,
 // where shifting characters by a certain value can obscure their meaning or reveal hidden messages.
 
+function isLetterOrNumber(char) {
+  return /^[a-zA-Z0-9]+$/.test(char);
```

Comment:
Some of this expression is not really doing anything for you for a single character (your 'char' name implies that's always the intended input). E.g. ^ is anchoring the beginning of a string, but a single char is by definition both the beginning and end. If you've done this consciously, and you're happy you know what ^, + and $ are doing here, fine. If you've found this as an expression online somewhere, I recommend having a quick look into what the various characters mean for your future benefit.

Reply:
Hi Dominic. I found a lot of regex on the internet, and I read more about it [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_expressions). If I understood correctly, the ^ is used to check the beginning of the string, + is used when I want to check one or more results, and the $ is used to check if it finds the desired character at the end of the string too.
I'm not sure if I need all of this or just /a-zA-Z0-9/. Still, I tried to make sure that the regex will check if the entire string consists of letters and/or numbers, with at least one occurrence.
But, being completely sincere, regex is still a bit tricky for me.

Thanks for your time and assistance.

Reply:
I thought that might be the case. Regex isn't tricky for you specifically; it's tricky for most people! It can get pretty complicated to the point that the pain of using regex is basically a meme for devs: https://xkcd.com/1171/

What you said as an explanation sounds good, though. In this case because it's just one character, parts of your regex don't have a functional effect, but it's good that you've at least looked up what the special characters mean. There's a whole world of pain in regex once you get to the more exotic end of things, but I wouldn't rush to learn it. There are much more important fundamentals to get your head round before you need to pay it too much attention.